source/MainPanel.cpp

```diff
@@ -50,6 +50,33 @@ using namespace std;
 
 
 
+namespace {
+	// Comparator class for comparing two strings, assumed to represent outfit
+	// category names, by the order of those categories in the UI.
+	class CategoryNameCompare {
+	public:
+		bool operator()(const std::string &a, const std::string &b)
+		{
+			if(a == b)
+				return false;
+
+			// Whichever is first in the array is considered smaller.
+			const vector<string> &outfitCategories = GameData::Category(CategoryType::OUTFIT);
+			for(const auto &it : outfitCategories)
+				if(it == a)
+					return true;
+				else if(it == b)
+					return false;
+
+			// Neither a nor b is a known category name.  Fall back
+			// to lexical comparison.
+			return (a < b);
```

Comment:
Is this line even reachable? Should it log a warning or something?

Reply:
if neither outfits have a category that is recorded it is, but then they should be "Special"
using a class for this one function seems overkill to me

Reply:
> using a class for this one function seems overkill to me

What would you suggest instead?

Reply:
Such items are already included in the scan. For instance, delete `category "Engines"` from all engines and scan something:

![2022-01-11-020737_268x262_scrot](https://user-images.githubusercontent.com/24784687/148868799-64fde6ab-a033-4fc8-8f25-f0f99c08c9eb.png)

I figure that blank line is because Outfit::Category returns an empty string in this case. I've changed that in d2fd2d995e2509c86f49c5f8bdeb645bf32d2cb8 so it prints "Unknown" instead of an empty string:

![2022-01-11-022320_269x288_scrot](https://user-images.githubusercontent.com/24784687/148870224-a733d808-2b13-4f08-b7ef-ef101b30a0fd.png)

(I didn't use "Special" because that's the title of the outfitter section for Local Map and licenses.)

The "Unknown" section is last in the scan results because it's an unknown string and the comparator considers unknown strings larger than known strings.

Note that in the case of outfits with no category, the `return (a < b)` line won't be reached. The only way to reach that line is to compare _two_ different category names that aren't in the game data. The empty string is _one_ such category name. If a ship had both an outfit with no category and an outfit with `category "Some made up string"`, _that_ would reach that line (and would sort the "Unknown" outfits before the "Some made up string" outfits).

But, anyway, the case of outfits with no category is now handled.